Add Keepers to the Documentation [Fixes #4477] (#4698)
ethereum/ethereum-org-website: src/content/developers/docs/oracles/index.md

}
```

### Chainlink Keepers {#chainlink-keepers}

> Smart contracts alone can't trigger or initiate their own functions at arbitrary times or under arbitrary conditions. State change will only occur when a transaction is initiated by another account (such as user, oracle, or contract). The [Chainlink Keeper Network](https://docs.chain.link/docs/chainlink-keepers/introduction/) provides options for smart contracts to outsource regular maintenance tasks in a trust minimized and decentralized manner.

Reviewer: ```suggestion
Smart contracts can't trigger or initiate their own functions at arbitrary times or under arbitrary conditions. State changes will only occur when another account initiates a transaction (such as a user, oracle, or contract). The [Chainlink Keeper Network](https://docs.chain.link/docs/chainlink-keepers/introduction/) provides options for smart contracts to outsource regular maintenance tasks in a trust minimized and decentralized manner.
```

> In order to use Chainlink Keepers, smart contract needs to implement [KeeperCompatibleInterface](https://docs.chain.link/docs/chainlink-keepers/compatible-contracts/) which consits of two functions:

Reviewer: ```suggestion
To use Chainlink Keepers, a smart contract must implement [KeeperCompatibleInterface](https://docs.chain.link/docs/chainlink-keepers/compatible-contracts/), which consists of two functions:
```

- `checkUpkeep` - Checks if the contract requires work to be done.
- `performUpkeep` - Performs the work on the contract, if instructed by checkUpkeep.

> The example below is a simple counter contract. The `counter` variable is incremented by one by every call to `performUpkeep`. [You can check out the following code using Remix](https://remix.ethereum.org/#url=https://docs.chain.link/samples/Keepers/KeepersCounter.sol)

Reviewer: ```suggestion
The example below is a simple counter contract. The `counter` variable is incremented by one by every call to `performUpkeep`. You can [check out the following code using Remix](https://remix.ethereum.org/#url=https://docs.chain.link/samples/Keepers/KeepersCounter.sol)
```

> The example below represents a simple counter contract. Each time performUpkeep is called, it increments its counter by one. [One can deploy the following code using Remix](https://remix.ethereum.org/#url=https://docs.chain.link/samples/Keepers/KeepersCounter.sol)

Reviewer: ```suggestion
The example below is a simple counter contract. The `counter` variable is incremented by one by every call to `performUpkeep`. [You can check out the following code using Remix](https://remix.ethereum.org/#url=https://docs.chain.link/samples/Keepers/KeepersCounter.sol)
```

// ./interfaces/KeeperCompatibleInterface.sol
import "@chainlink/contracts/src/v0.7/KeeperCompatible.sol";

contract Counter is KeeperCompatibleInterface {
    /**
    * Public counter variable
    */
    uint public counter;

    /**
    * Use an interval in seconds and a timestamp to slow execution of Upkeep
    */
    uint public immutable interval;
    uint public lastTimeStamp;

    constructor(uint updateInterval) {
      interval = updateInterval;
      lastTimeStamp = block.timestamp;

      counter = 0;
    }

    function checkUpkeep(bytes calldata /* checkData */) external override returns (bool upkeepNeeded, bytes memory /* performData */) {
        upkeepNeeded = (block.timestamp - lastTimeStamp) > interval;
        // We don't use the checkData in this example. The checkData is defined when the Upkeep was registered.
    }

    function performUpkeep(bytes calldata /* performData */) external override {
        lastTimeStamp = block.timestamp;
        counter = counter + 1;
        // We don't use the performData in this example. The performData is generated by the Keeper's call to your checkUpkeep function
    }
}
```

> After one deploys a contract, it does not automatically begin to receive Upkeeps. The final step is [Upkeep Registration](https://docs.chain.link/docs/chainlink-keepers/register-upkeep/).

Reviewer: ```suggestion
After deploying a contract, it does not automatically begin to receive Upkeeps. The final step is [Upkeep Registration](https://docs.chain.link/docs/chainlink-keepers/register-upkeep/).
```

Reviewer: This feels a little incomplete to me. At the very least we should explicitly state what `receiving an Upkeep` means here instead of expecting the reader to deduce it from the surrounding context.

Author: Hey @minimalsm, thanks for the feedback! You are right, this is a little misleading. How does the next sentence sound to you?

> After deploying a Keeper-compatible contract, you must register it for [Upkeep](https://docs.chain.link/docs/chainlink-keepers/register-upkeep/).

Reviewer: @andrejrakic yeah this sounds better but I'd explain why.

> After deploying a Keeper-compatible contract, you must register the contract for Upkeep to....

Author: @minimalsm good point! Do you think the following sentence is clear to the average reader?
> After deploying a Keeper-compatible contract, you must register the contract for [Upkeep](https://docs.chain.link/docs/chainlink-keepers/register-upkeep/) and fund it with LINK, to notify the Keeper Network about your contract, so your work is performed continuously.

> Other Keepers projects:

Reviewer: ```suggestion
### Keepers projects {#keepers}
```
